Automattic/wp-calypso
# Plugins: Reduxify, cleanup and modernize notices

This PR reduxifies, cleans up and modernizes the `PluginNotices` component. It grew rather a bit larger than I was hoping, but it's really a huge step for it and changes it to a state where we can just leave it be. 

It may make a lot of sense to review different commits separately, as changes make more sense that way. I'll also do a self-review in hope that it'll make the PR easier to review.

This PR is part of #24180.

#### Changes proposed in this Pull Request

* Update plugin `filter*` utility methods (and their tests) to work with site IDs and plugin IDs.
* Update `PluginNotices` to use Redux for retrieving and managing notices. This requires a lot of changes because of the different ways we're now storing those in Redux (for example, site IDs and not site objects; plugin IDs and not plugin objects).
* Update `SinglePlugin` to pass plugins and sites to `PluginNotices`
* Update `PluginsList` to pass plugins and sites to `PluginNotices`
* Dispatch Redux actions directly instead of using the redux bridge
* Use Redux for notices instead of the notices library.
* Use `localize()` instead of raw `i18n.translate` to subscribe to eventual locale changes

#### Testing instructions

* Get several Jetpack sites with some plugins for testing. Ideally, you would also have a connected multisite with 1-2 subsites that are also connected too. 
* Thoroughly test notices in all plugin scenarios; compare against production and verify that global and inline notices in plugin items still work the same way when updating, installing, or removing plugins, as well as toggling auto-updates and activating/deactivating plugins:
  * `/plugins/:plugin/:site`
  * `/plugins/:plugin`
  * `/plugins`
  * `/plugins/manage`
  * `/plugins/manage` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
  * `/plugins/manage/:site` - for a single plugin
  * `/plugins/manage/:site` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
* Verify all tests pass.
